### ISSUE TITLE
Error to generate the cifti files

### ISSUE THREAD

**Issue by xuyangwen** [poldracklab/fmriprep#1810]
Hello,

I have a problem to generate the cifti files. It seems that this error does not occur when I did not put fsaverage6 into the following command line.

fMRIPrep version: 1.5.0

fMRIPrep command: /usr/local/miniconda/bin/fmriprep /data /out participant --participant-label sub-02 --ignore fieldmaps slicetiming --output-spaces MNI152NLin2009cAsym MNI152NLin6Asym:res-2 fsaverage5 fsaverage6 --use-aroma --dummy-scans 10 --cifti-output --write-graph -w /scratch

Here is the error information:

```
Node: fmriprep_wf.single_subject_02_wf.func_preproc_task_scrambledNarrative2_wf.gen_cifti
Working directory: /scratch/fmriprep_wf/single_subject_02_wf/func_preproc_task_scrambledNarrative2_wf/gen_cifti

Node inputs:

TR = 1.0
bold_file = <undefined>
gifti_files = [['/scratch/fmriprep_wf/single_subject_02_wf/func_preproc_task_scrambledNarrative2_wf/bold_surf_wf/update_metadata/mapflow/_update_metadata0/lh.fsaverage5.gii', '/scratch/fmriprep_wf/single_subject_02_wf/func_preproc_task_scrambledNarrative2_wf/bold_surf_wf/update_metadata/mapflow/_update_metadata1/lh.fsaverage6.gii', '/scratch/fmriprep_wf/single_subject_02_wf/func_preproc_task_scrambledNarrative2_wf/bold_surf_wf/update_metadata/mapflow/_update_metadata2/rh.fsaverage5.gii', '/scratch/fmriprep_wf/single_subject_02_wf/func_preproc_task_scrambledNarrative2_wf/bold_surf_wf/update_metadata/mapflow/_update_metadata3/rh.fsaverage6.gii']]
subjects_dir = <undefined>
surface_target = ['fsaverage5', 'fsaverage6']
volume_target = MNI152NLin2009cAsym

Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/multiproc.py", line 269, in _send_procs_to_workers
    num_subnodes = self.procs[jobid].num_subnodes()
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 1205, in num_subnodes
    self._check_iterfield()
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 1239, in _check_iterfield
    "have the same length. %s") % str(self.inputs))
ValueError: All iterfields of a MapNode have to have the same length. 
TR = 1.0
bold_file = <undefined>
gifti_files = [['/scratch/fmriprep_wf/single_subject_02_wf/func_preproc_task_scrambledNarrative2_wf/bold_surf_wf/update_metadata/mapflow/_update_metadata0/lh.fsaverage5.gii', '/scratch/fmriprep_wf/single_subject_02_wf/func_preproc_task_scrambledNarrative2_wf/bold_surf_wf/update_metadata/mapflow/_update_metadata1/lh.fsaverage6.gii', '/scratch/fmriprep_wf/single_subject_02_wf/func_preproc_task_scrambledNarrative2_wf/bold_surf_wf/update_metadata/mapflow/_update_metadata2/rh.fsaverage5.gii', '/scratch/fmriprep_wf/single_subject_02_wf/func_preproc_task_scrambledNarrative2_wf/bold_surf_wf/update_metadata/mapflow/_update_metadata3/rh.fsaverage6.gii']]
subjects_dir = <undefined>
surface_target = ['fsaverage5', 'fsaverage6']
volume_target = MNI152NLin2009cAsym
```

Does anybody know how to solve this?

Thanks!

**Comment by mgxd** [comment]
@xuyangwen this looks like a bug - i think you can avoid this if you specify only one surface output-space (fsavg 5/6) at a time.

**Comment by xuyangwen** [comment]
> 
> 
> @xuyangwen this looks like a bug - i think you can avoid this if you specify only one surface output-space (fsavg 5/6) at a time.

Thanks!

**Comment by oesteban** [comment]
@mgxd how are we dealing with this in the context of #1887?

**Comment by mgxd** [comment]
this shouldn't be a problem in #1887, since we'll be dropping fsaverage surface support

**Comment by oesteban** [comment]
A fix for this has been addressed in #1955 and will be rolled out with version 20.0.0rc1. Please reopen if this regresses.
